src/store-credit/store-credit-action-creator.ts

```diff
@@ -0,0 +1,39 @@
+import { createAction, createErrorAction, ThunkAction } from '@bigcommerce/data-store';
+import { Observable, Observer } from 'rxjs';
+
+import { InternalCheckoutSelectors } from '../checkout';
+import { MissingDataError, MissingDataErrorType } from '../common/error/errors';
+import { RequestOptions } from '../common/http-request';
+
+import { ApplyStoreCreditAction, StoreCreditAction, StoreCreditActionType } from './store-credit-actions';
+import StoreCreditRequestSender from './store-credit-request-sender';
+
+export default class StoreCreditActionCreator {
+    constructor(
+        private _storeCreditRequestSender: StoreCreditRequestSender
+    ) {}
+
+    applyStoreCredit(useStoreCredit: boolean, options?: RequestOptions): ThunkAction<StoreCreditAction, InternalCheckoutSelectors> {
+        return store => Observable.create(async (observer: Observer<ApplyStoreCreditAction>) => {
```

Comment:
This comment is completely optional as it's simply a different way of writing the same thing. But lately, we've been using observable operators + factories instead of creating observables directly using `new` or `Observable.create`. For example, your code would look something like this instead: 

```ts
return store => (
    concat(
        of(createAction(StoreCreditActionType.ApplyStoreCreditRequested)),
        defer(async () => {
            const { checkout: { getCheckout } } = store.getState();

            if (!checkout) {
                throw new MissingDataError(MissingDataErrorType.MissingCheckout);
            }

            const { body } = await useStoreCredit ?
                this._storeCreditRequestSender.applyStoreCredit(checkout.id, options)
                this._storeCreditRequestSender.removeStoreCredit(checkout.id, options);
            
            return createAction(StoreCreditActionType.ApplyStoreCreditSucceeded, body);
        })
    ).pipe(
        catchError(error => throwErrorAction(StoreCreditActionType.ApplyStoreCreditFailed, error))
    );
);
```

In your case, it doesn't really make a difference. But in some other cases where we have to chain or combine two or more actions, operators make it easier to achieve (as you can just insert another observable in `concat`). 🍺